tvtk/pyface/tvtk_scene.py

```diff
@@ -228,6 +229,14 @@ def __setstate__(self, str_state):
         state_pickler.set_state(self, state_pickler.loads_state(str_state))
 
     ###########################################################################
+    # 'event' interface.
+    ###########################################################################
+    def _closed_fired(self):
+        del self.light_manager
+        del self._interactor
+        del self
```

Comment:
I think for traits it is better to do.

```python
self.light_manager = None
self._interactor = None
```

Also there should not be a need to `del self` since `self` will be garbadge collected as soon as the method is called,